python/lsst/daf/butler/core/dimensions/_coordinate.py

```diff
@@ -1045,3 +1049,14 @@ def hasRecords(self) -> bool:
     def _record(self, name: str) -> Optional[DimensionRecord]:
         # Docstring inherited from DataCoordinate.
         return self._records[name]
+
+    def __getattr__(self, name: str) -> Any:
+        try:
+            return self._record(name)
+        except KeyError:
+            raise AttributeError(name) from None
```

Comment:
It wouldn't hurt to also have a test on an unexpanded dataId to make sure it raises AttributeError? Then I imagine the follow up will be that people will want to distinguish between an attribute that is failing because it's not part of the dataId vs one that is failing because the records were not available (so maybe the error message should mention the distinction).